drivers/gpio/gpio_litex.c

```diff
@@ -191,17 +189,49 @@ static int gpio_litex_port_toggle_bits(const struct device *dev,
 	return 0;
 }
 
+static void gpio_litex_irq_handler(const struct device *dev)
+{
+	const struct gpio_litex_cfg *gpio_config = DEV_GPIO_CFG(dev);
+	struct gpio_litex_data *data = dev->data;
+
+	uint8_t int_status =
+		litex_read(GPIO_EV_PENDING(dev), gpio_config->reg_size);
+
+	/* clear events */
+	litex_write(GPIO_EV_PENDING(dev), gpio_config->reg_size,
+			int_status);
+
+	gpio_fire_callbacks(&data->cb, dev, int_status);
+}
+
+static int gpio_litex_manage_callback(const struct device *dev,
+				      struct gpio_callback *callback, bool set)
+{
+	struct gpio_litex_data *data = dev->data;
+
+	return gpio_manage_callback(&data->cb, callback, set);
+}
+
 static int gpio_litex_pin_interrupt_configure(const struct device *dev,
 					      gpio_pin_t pin,
 					      enum gpio_int_mode mode,
 					      enum gpio_int_trig trig)
 {
-	int ret = 0;
+	const struct gpio_litex_cfg *gpio_config = DEV_GPIO_CFG(dev);
+
+	if (gpio_config->port_is_output == true) {
+		return -ENOTSUP;
+	}
 
-	if (mode != GPIO_INT_MODE_DISABLED) {
-		ret = -ENOTSUP;
+	if (mode == GPIO_INT_MODE_EDGE) {
+		uint8_t ev_enabled = litex_read(GPIO_EV_ENABLE(dev),
+				gpio_config->reg_size);
+		litex_write(GPIO_EV_ENABLE(dev), gpio_config->reg_size,
+			    ev_enabled | BIT(pin));
+		return 0;
 	}
-	return ret;
+
+	return -ENOTSUP;
```

Comment:
This function should provide a means to disable an active interrupt by passing `GPIO_INT_DISABLE` as the mode.

Reply:
Thanks for pointing it oout. I've added support for disabling interrupts in latest commit. It has been tested on Arty A7 with LiteX SoC and it works great.

Reply:
@madscientist159 thanks for spotting.